src/main/scala/io/iohk/ethereum/blockchain/sync/SyncController.scala

```diff
@@ -71,18 +71,15 @@ class SyncController(
   }
 
   def handlePeerUpdates: Receive = {
-    case peers: PeerManagerActor.Peers =>
-      peers.peers.foreach {
-        case (peer, _: PeerActor.Status.Handshaked) =>
-          if (!handshakedPeers.contains(peer)) context watch peer.ref
-
-        case (peer, _) if handshakedPeers.contains(peer) =>
-          removePeer(peer.ref)
+    case PeersInfoHolderActor.HandshakedPeers(peers) =>
+      peers.foreach {
+        case (peer, _) if !handshakedPeers.contains(peer) =>
+          context watch peer.ref
 
         case _ => // nothing
       }
 
-      handshakedPeers = peers.handshaked
+      handshakedPeers = peers
 
     case Terminated(ref) if handshakedPeers.exists(_._1.ref == ref) =>
       removePeer(ref)
```

Comment:
This is a little bit confusing to me. We are adding peers based on `PeersInfoHolderActor` but removing them because we are watching them. Maybe we need another opinion.